ui/lib/sync/addon/components/secrets/page/destinations.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { service } from '@ember/service';
+import { action } from '@ember/object';
+import { getOwner } from '@ember/application';
+import errorMessage from 'vault/utils/error-message';
+import { findDestination } from 'core/helpers/sync-destinations';
+
+import type SyncDestinationModel from 'vault/vault/models/sync/destination';
+import type RouterService from '@ember/routing/router-service';
+import type StoreService from 'vault/services/store';
+import type FlashMessageService from 'vault/services/flash-messages';
+import type { EngineOwner } from 'vault/vault/app-types';
+import type { SyncDestinationName, SyncDestinationType } from 'vault/vault/app-types';
+
+interface Args {
+  destinations: Array<SyncDestinationModel>;
+  nameFilter: SyncDestinationName;
+  typeFilter: SyncDestinationType;
+}
+
+export default class SyncSecretsDestinationsPageComponent extends Component<Args> {
+  @service declare readonly router: RouterService;
+  @service declare readonly store: StoreService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  // typeFilter arg comes in as destination type but we need to pass the name of the type into the SearchSelect
+  get typeFilter() {
+    return findDestination(this.args.typeFilter)?.name;
+  }
+
+  get destinationNames() {
+    return this.args.destinations.map((destination) => ({ id: destination.name, name: destination.name }));
+  }
+
+  get destinationTypes() {
+    return this.args.destinations.reduce((types: Array<{ id: string; name: string }>, destination) => {
+      const { typeDisplayName } = destination;
+      const isUnique = !types.find((type) => type.name === typeDisplayName);
+      if (isUnique) {
+        types.push({ id: typeDisplayName, name: destination.type });
+      }
+      return types;
+    }, []);
+  }
+
+  get mountPoint(): string {
+    const owner = getOwner(this) as EngineOwner;
+    return owner.mountPoint;
+  }
+
+  get paginationQueryParams() {
+    return (page: number) => ({ page });
+  }
+
+  get noResultsMessage() {
+    const { nameFilter, typeFilter } = this.args;
+    if (nameFilter && typeFilter) {
+      return `There are no ${typeFilter} destinations matching "${nameFilter}".`;
+    }
+    if (nameFilter) {
+      return `There are no destinations matching "${nameFilter}".`;
+    }
+    if (typeFilter) {
+      return `There are no ${typeFilter} destinations.`;
+    }
+    return '';
+  }
+
+  @action
+  onFilterChange(key: string, selectObject: Array<{ id: string; name: string } | undefined>) {
+    this.router.transitionTo('vault.cluster.sync.secrets.destinations', {
+      queryParams: { [key]: selectObject[0]?.name },
+    });
+  }
+
+  @action
+  async onDelete(destination: SyncDestinationModel) {
```

Comment:
I wonder if we should make this a concurrency task? Purging a destination of all synced secrets might take some time, leveraging the loading state for this delete action could be useful. 

Though, a long loading state in the confirm action may be a little strange so design may want to have different ways of showing the delete progress

Reply:
I think that's a good idea but something we should defer until after we update this pattern to show a modal. Right now the confirm closes on click so we can't really keep the user from doing other things while we wait for the request to resolve.

Reply:
Yeah - one thought I had was adding a flash that said "Deleting destination, unsyncing secrets" and then showing the success when deletion was complete

Reply:
Also a good idea but we would need to handle that outside of the component since it would be torn down if the user moves to a different route before the request resolves. Maybe we could fire the flash message from the adapter?

Since there is a task to handle the delete action in the toolbar let's defer this and discuss further in the scope of that work. I'll add a comment to the ticket.